apps/client/app/models/user.server.ts

```diff
@@ -0,0 +1,76 @@
+import { User } from '~/types/auth';
+
+// Simulate a user database
+// TODO: Replace this with an API call to a real database
+const usersMap = new Map<string, User>();
+
+export function storeUser(user: User) {
+  usersMap.set(user.id, user);
+  return user.id;
+}
+
+export function getUserById(userId: string | undefined) {
+  if (!userId) {
+    return undefined;
+  }
+
+  return usersMap.get(userId);
+}
+
+/**
+ * Retrieves a user based on the specified attribute and value.
+ * @param attribute - The attribute of the user object to search for.
+ * @param value - The value to match against the specified attribute.
+ * @returns The user object if found, or null if not found.
+ */
+export function getUserByAttribute(attribute: keyof User, value: unknown) {
+  for (const user of usersMap.values()) {
+    if (user[attribute] === value) {
+      return user;
+    }
+  }
+  return null;
+}
+
+// In-Source test suites
+// https://vitest.dev/guide/in-source.html
```

Comment:
Im not understand this, this file seems to be a source code file, and below is the code that looks unit test, is this right?

Reply:
This is a feature of Vitest, see [this.](https://vitest.dev/guide/in-source.html) This is useful because I can access `usersMap`, which is not exported, to verify the methods are working correctly

Reply:
I see that could have some advantages, but honestly, i think that will make the code a mess, mix unit tests and source, but this is my opinion i think that this could be discussed the rest after we have the most priority things done.
Ty